.changeset/authenticatedclient_no_longer_inherits_from_client.md

```diff
@@ -0,0 +1,7 @@
+---
+default: major
+---
+
+#### `AuthenticatedClient` no longer inherits from `Client`
+
+The API of `AuthenticatedClient` is still a superset of `Client`, but the two classes no longer share a common base class.
```

Comment:
It would be nice to have an `ABC` or `typing.Protocol`, which would be used in the signature types when expecting a client. This would more easily allow for custom clients, e.g. with a different authentication mechanism besides bearer-tokens. Currently, this would require inheriting from eiter `Client` or `AuthenticatedClient` to pass static type-checking.

Reply:
I 100% agree, but I think this should be done with a bigger overhaul of authentication via #152 (and similar). The intention of the typing right now is to indicate _which_ endpoints require authentication and which don't. There is no protocol difference between the two (that the endpoint functions care about), so all we have is the separate types right now.